nemo/collections/nlp/data/glue_benchmark/glue_benchmark_dataset.py

```diff
@@ -347,6 +358,118 @@ def _truncate_seq_pair(self, tokens_a: str, tokens_b: str, max_length: int):
+
+    def __len__(self):
+        return len(self.examples)
+
+    def __getitem__(self, idx):
+        enc_query, dec_input, labels = self.features[idx]
+        return {'text_enc': enc_query, 'text_dec': dec_input, 'labels': labels}
+
+    def collate_fn(self, batch):
+        enc_query = [item['text_enc'] for item in batch]
+        dec_input = [item['text_dec'] for item in batch]
+        labels = [item['labels'] for item in batch]
+
+        max_dec_input_length = max([len(item) for item in dec_input]) if dec_input else 0
+        max_enc_query_length = max([len(item) for item in enc_query]) if enc_query else 0
+        max_label_length = max([len(item) for item in labels]) if labels else 0
+
+        loss_mask = [([1] * (len(item))) + ([0] * (max_label_length - len(item))) for item in labels]
+        enc_query = [item + [self.tokenizer.pad_id] * (max_enc_query_length - len(item)) for item in enc_query]
+        dec_input = [item + [self.tokenizer.pad_id] * (max_dec_input_length - len(item)) for item in dec_input]
+        labels = [item + [self.tokenizer.pad_id] * (max_label_length - len(item)) for item in labels]
+
+        enc_query = torch.LongTensor(enc_query)
+        dec_input = torch.LongTensor(dec_input)
+        labels = torch.LongTensor(labels)
+        loss_mask = torch.LongTensor(loss_mask)
+
+        enc_mask = make_attention_mask_3d(enc_query, enc_query, self.tokenizer.pad_id).long()
+        dec_mask = make_attention_mask_3d(dec_input, dec_input, self.tokenizer.pad_id)
+        dec_mask = (dec_mask * make_history_mask_3d(dec_input)).long()
+        enc_dec_mask = make_attention_mask_3d(dec_input, enc_query, self.tokenizer.pad_id).long()
+
+        return {
+            'text_enc': enc_query,
+            'text_dec': dec_input,
+            'labels': labels,
+            'loss_mask': loss_mask,
+            'enc_mask': enc_mask,
+            'dec_mask': dec_mask,
+            'enc_dec_mask': enc_dec_mask,
+        }
+
+    def make_history_mask_3d(self, block):
+        batch, length = block.shape
+        arange = np.arange(length)
+        history_mask = (arange[None,] <= arange[:, None])[
+            None,
+        ]
+        history_mask = np.repeat(history_mask, batch, 0)
+        return history_mask
+
+    def convert_examples_to_features(self):
+        """
+        Converts examples into Text-to-Text batches to be used with a model like T5.
+        Inputs are prefixed with a text prompt that indicates the task to perform.
+        """
+        features = []
+        for ex_index, example in enumerate(self.examples):
+            if ex_index % 10000 == 0:
+                logging.info(f"Writing example {ex_index} of {len(self.examples)}")
+
+            text_to_text_query = self.processor.get_t5_prompted_query(example.text_a, example.text_b)
+            enc_query = self.tokenizer.text_to_ids(text_to_text_query)
+            if len(enc_query) > self.max_seq_length:
```

Comment:
The GLUE dataset restricts the examples to have `text_a`, `text_b` only where the T5 is more generic than that. We can use the dataset like the one in the GPT p-tuning PR, which can handle any fields in the data for different tasks.